pkg/controller/appdefinition/deploy.go

```diff
@@ -504,6 +507,14 @@ func addImageAnnotations(annotations map[string]string, appInstance *v1.AppInsta
 	}
 }
 
+func addPrometheusAnnotations(annotations map[string]string, container v1.Container) {
+	if container.Metrics.Path != "" && container.Metrics.Port != 0 {
+		annotations[labels.PrometheusScrape] = "true"
+		annotations[labels.PrometheusPath] = container.Metrics.Path
+		annotations[labels.PrometheusPort] = strconv.Itoa(int(container.Metrics.Port))
+	}
```

Comment:
Trying to think through the edge cases from a UX perspective: How is a user informed when they create a subtly invalid combination like `"", 80` or `"/", 0"` that fails this check? What about something blatantly invalid that passes this check; e.g. `"/", -3`?

Is there validation somewhere else (in the AML schema maybe)?

Reply:
Hmm that is a very good point. I definitely need to prevent negative ports from happening, and should probably set the upper bound at 65535. I should be able to enforce certain regex in the AML scheme, so I'll try that out.

Reply:
I created a PR for the AML scheme that does better validation here: https://github.com/acorn-io/aml/pull/8

Reply:
Nice!

at some point, it would probably also be wise to have validation in the k8s API too, but that's not something I'd push for now.